### ISSUE TITLE
Compute-Intensive Kernel Hits NaN

### ISSUE THREAD

**Issue by frobnitzem** [StanfordLegion/task-bench#71]
This kernel:
https://github.com/StanfordLegion/task-bench/blob/e24115266092e553e28722980b0ae8f4f4f4827a/core/core_kernel.cc#L234
repeatedly applies
A = A*A + A (i.e. fmadd(A, A, A))
which quickly escalates to A=NaN.

To avoid this, the kernel can be changed to
A = -A*A + A (i.e. fnmsub(A, A, A)).

This makes the iteration equivalent to the Logistic map with r=1.
Adjusting the initial condition to A = 0.7 (or anything between 0 and 1) makes the iterations converge slowly to 0 over time.

**Comment by hyviquel** [comment]
@elliottslaughter Do you think this problem can be linked to #69 ?

**Comment by elliottslaughter** [comment]
@hyviquel It shouldn't have anything to do with that issue since the results of the compute bound kernel are thrown away, and the "result" that is actually written into the output region is instead a tuple containing the timestep and point (column) of the task.

@frobnitzem Thanks for pointing this out. Overall this looks good to me. I don't think it will change the practical results on any platforms we tested (since we already verified that we hit peak FLOPS), but it is true that a future system might hypothetically add an early-out for NaNs (which would then cause Task Bench to over-report its achieved FLOPS).

I'm happy to take a PR on this or may get back to it myself in a week or so. (Currently digging myself out of things that have been piling up since the break.)

**Comment by frobnitzem** [comment]
I agree, my timings were the same after changing the code. I didn't make a
PR because one of the avx cases doesn't have a clear fix.
